src/js/internal/fs/glob.ts

```diff
@@ -1,58 +1,75 @@
-  exclude(ent: string): boolean;
-}
-
 async function* glob(pattern: string | string[], options?: GlobOptions): AsyncGenerator<string> {
-  pattern = validatePattern(pattern);
+  const patterns = validatePattern(pattern);
   const globOptions = mapOptions(options || {});
-  let it = new Bun.Glob(pattern).scan(globOptions);
   const exclude = globOptions.exclude;
+  const excludeGlobs = Array.isArray(exclude) ? exclude.map(p => [new Bun.Glob(p), p] as const) : null;
 
-  for await (const ent of it) {
-    if (exclude(ent)) continue;
-    yield ent;
+  for (const pat of patterns) {
+    for await (const ent of new Bun.Glob(pat).scan(globOptions)) {
+      const _ent = ent.replaceAll(sep, "/");
+      if (typeof exclude === "function" && exclude(ent)) continue;
+      if (excludeGlobs?.some(([glob, p]) => glob.match(ent) || _ent === p || _ent.startsWith(p + "/"))) continue;
+      yield ent;
+    }
   }
 }
 
 function* globSync(pattern: string | string[], options?: GlobOptions): Generator<string> {
-  pattern = validatePattern(pattern);
+  const patterns = validatePattern(pattern);
   const globOptions = mapOptions(options || {});
-  const g = new Bun.Glob(pattern);
   const exclude = globOptions.exclude;
-  for (const ent of g.scanSync(globOptions)) {
-    if (exclude(ent)) continue;
-    yield ent;
+  const excludeGlobs = Array.isArray(exclude) ? exclude.map(p => [new Bun.Glob(p), p] as const) : null;
+
+  for (const pat of patterns) {
+    for (const ent of new Bun.Glob(pat).scanSync(globOptions)) {
+      const _ent = ent.replaceAll(sep, "/");
+      if (typeof exclude === "function" && exclude(ent)) continue;
+      if (excludeGlobs?.some(([glob, p]) => glob.match(ent) || _ent === p || _ent.startsWith(p + "/"))) continue;
+      yield ent;
+    }
   }
 }
 
-function validatePattern(pattern: string | string[]): string {
-  if ($isArray(pattern)) {
-    throw new TypeError("fs.glob does not support arrays of patterns yet. Please open an issue on GitHub.");
+function validatePattern(pattern: string | string[]): string[] {
+  if (typeof pattern === "string") {
+    validateString(pattern, "pattern");
+    return [isWindows ? pattern.replaceAll("/", sep) : pattern];
   }
-  validateString(pattern, "pattern");
-  return isWindows ? pattern.replaceAll("/", "\\") : pattern;
+  validateArray(pattern, "pattern");
+  return pattern.map(p => {
+    validateString(p, "pattern");
+    return isWindows ? p.replaceAll("/", sep) : p;
+  });
 }
 
-function mapOptions(options: GlobOptions): ExtendedGlobOptions {
+function mapOptions(options: GlobOptions): GlobScanOptions & { exclude: GlobOptions["exclude"] } {
   validateObject(options, "options");
 
-  const exclude = options.exclude ?? no;
-  validateFunction(exclude, "options.exclude");
+  let exclude = options.exclude ?? no;
+  if (Array.isArray(exclude)) {
+    validateArray(exclude, "options.exclude");
+    if (isWindows) {
+      exclude = exclude.map((pattern: string) => pattern.replaceAll("\\", "/"));
```

Comment:
Shouldn't this be the other way around?

```typescript
pattern.replaceAll("/", "\\")
```

or:

```typescript
pattern.replaceAll("/", sep)
```

Reply:
If so, we should add a test for this (passing in an `exclude` pattern that uses "/" separators on Windows).

Reply:
Bun.Glob.match was being weird and only seemed to work with the normal separators (you can see my commit history to see)

Reply:
I also have a test that uses the slash which is how I found the issue
```ts
    const exclude2 = ["folder.test/another-folder"];
    const expected2 = isWindows ? ["folder.test\\file.txt"] : ["folder.test/file.txt"];
```

Reply:
Ah, I see. Yeah, `Bun.Glob` only wants `/` separators in the pattern. I thought simply replacing `\` with `/` would break matching for UNC/DOS paths, but it seems to work.